lib/pages/profile_dialog.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import '../gen/assets.gen.dart';
+import '../l10n/l10n.dart';
+import '../widgets/options_list_item.dart';
+import 'api_key_dialog.dart';
+
+class ProfileDialog extends StatefulWidget {
+  @override
+  _ProfileDialogState createState() => _ProfileDialogState();
+}
+
+class _ProfileDialogState extends State<ProfileDialog> {
+  bool loggedIn = false;
+  String logText = L10nStrings.profileDialog_logIn;
+  Icon logIcon = const Icon(Icons.login);
```

Comment:
These should not be held in the state. These can be derived from `loggedIn` and thus should be conditionally rendered in the UI